### DISCUSSION TITLE
Get validation value in errors object

### DISCUSSION THREAD

**Discussion by JPortegijs** [react-hook-form/react-hook-form#3725] [Q&A]
Hello,

I'm trying to make some sort of i18n error message library and I ran into this problem.

Is there a way to get the value of a validation rule inside the error object? I would like to do something like this:

`rules: {
 required: true,
 minLength: 4,
},`

And then use minLength value in my error message, (string replace on %s):
<img width="238" alt="image" src="https://user-images.githubusercontent.com/12064206/102600753-8a98e680-411f-11eb-85a1-e7ca189277d6.png">

Any help is appreciated.

**Comment by bluebill1049** [comment]
have you tried `value` and `message` with rules.

**Reply by JPortegijs** [reply]
Yes, I tried doing it like:

`rules: {
 minLength: {
  value: 4,
 },
}`

But value is still not accessible from the errors object. When rendering the error I need to replace %s with 4.

**Reply by markivancho** [reply]
I guess it's not possible to get `someValue` from `errors` object, probably you have to inject it into the error message.
```
    rules: {
      minLength : {
        value: someValue,
        message: `This field requires at least ${someValue} characters`
      }
    }
```

**Reply by bluebill1049** [reply]
Hmm maybe you need to have a function to return that rule value.

```
const getRule =  () => {
  const value = 4;
  
  return {
    value, 
    message: 'xxxx' + value
  }
}

rules: getRule();
```
